functions/backup.bash

```diff
@@ -237,9 +215,156 @@ amanda_setup() {
+##
+mirror_SD() {
+  local src
+  local dest
+  local start
+  local syncMount="/storage/syncmount"
+
+
+  src=/dev/mmcblk0
+  if [[ -n "$INTERACTIVE" ]]; then
+    select_blkdev "^sd" "Setup SD mirroring" "Select the USB attached disk device to copy the internal SD card data to"
+    # shellcheck disable=SC2154
+    if [[ -z "$retval" ]]; then return 0; fi
+    dest="/dev/$retval"
+  else
+    dest="${backupdrive:-/dev/sda}"
+  fi
+  if [[ "$1" == "raw" ]]; then
+    echo "Creating a raw partition copy, be prepared this may take long such as 20-30 minutes for a 16 GB SD card"
+    if ! cond_redirect dd if="${src}" bs=1M of="${dest}"; then echo "FAILED (raw device copy)"; return 1; fi
+    echo "OK"
+    return 0;
   fi
 
+  if [[ "$1" == "diff" ]]; then
+    if [[ ! -d $syncMount ]]; then
+      mkdir -p "$syncMount"
+    fi
+    if [[ -n "$INTERACTIVE" ]]; then
+      select_blkdev "^-sd" "select partition" "Select the partition to copy the internal SD card data to"
+      dest="/dev/$retval"
+    else
+      dest=${dest}2
+    fi
+    # Auswahl des src dir nötig ?? nein immer "/"
+    # Auswahl des dest dir nötig ?? nein immer syncmount und dest wird oben ausgewählt
+    #if [[ -n "$INTERACTIVE" ]]; then
+    mount "$dest" "${syncMount:-/storage/syncmount}"
+    rsync --one-file-system -avRh "/" "$syncMount"
+    if ! umount "$syncMount" &> /dev/null; then
+      sleep 1
+      umount -l "$syncMount" &> /dev/null
+    fi
+  fi
+}
+
+
+## setup mirror/sync of boot and / partitions
+##
+##   setup_mirror_SD()
+##
+setup_mirror_SD() {
+  local dest
+  local srcSize
+  local destSize
+  local minSize
+  local targetDir="/etc/systemd/system/"
+  local storageDir=${storagedir:-/storage}
+  local sizeError="your destination SD card device does not have enough space, it needs to have at least twice as much as the source"
+  local infoText1="DANGEROUS OPERATION, USE WITH PRECAUTION!\\n\\nThis will *copy* your system root from your SD card to a USB attached card writer device. Are you sure"
+  local infoText2="is an SD card writer device equipped with a dispensible SD card ? Are you this will destroy all data on that card and you want to proceed writing to this device ?"
+
+
+  if [[ "$1" == "remove" ]]; then
+    rm -f ${targetDir}/sdr*.{service,timer}
+    cond_redirect systemctl -q daemon-reload &> /dev/null
+    return 0
+  fi
+  if [[ "$1" != "install" ]]; then echo "FAILED"; return 1; fi
+
+  if ! is_pi; then
+    if [[ -n "$INTERACTIVE" ]]; then
+      whiptail --title "Incompatible hardware detected" --msgbox "Mirror SD: this option is for the Raspberry Pi only." 10 60
+    fi
+    echo "FAILED"; return 1
+  fi
+
+  # shellcheck disable=SC2154
+  if [[ -n "$UNATTENDED" ]] && [[ -z "$backupdrive" ]]; then return 0; fi
+
+  if [[ -n "$INTERACTIVE" ]]; then
+    select_blkdev "^sd" "Setup SD mirroring" "Select USB device to copy the internal SD card data to"
+    if [[ -z "$retval" ]]; then return 0; fi
+    dest="/dev/$retval"
```

Comment:
Use `{}` in a variable assignment involving another variable to avoid ambiguity.

Reply:
ok